plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2933,6 +2937,32 @@ private void finishWrite(ConnectorSession session, IcebergTableHandle table, Col
         commitUpdateAndTransaction(rowDelta, session, transaction, "write");
     }
 
+    private TupleDomain<IcebergColumnHandle> extractTupleDomainsFromCommitTasks(Table icebergTable, List<CommitTaskData> commitTasks)
+    {
+        List<IcebergColumnHandle> partitionColumns = getProjectedColumns(icebergTable.schema(), typeManager, identityPartitionColumnsInAllSpecs(icebergTable));
+        Type[] partitionColumnTypes = icebergTable.spec().fields().stream()
+                .map(field -> field.transform().getResultType(
+                        icebergTable.schema().findType(field.sourceId())))
+                .toArray(Type[]::new);
+
+        Map<IcebergColumnHandle, Domain> map = new HashMap<>();
+        for (CommitTaskData commitTask : commitTasks) {
+            if (commitTask.partitionDataJson().isPresent()) {
+                PartitionData partitionData = PartitionData.fromJson(commitTask.partitionDataJson().get(), partitionColumnTypes);
+                Map<Integer, Optional<String>> partitionKeys = getPartitionKeys(partitionData, icebergTable.spec());
```

Comment:
Can be extracted outside of the `for` statement.

Reply:
Keys are extracted for each `commitTask` and can be different, for example after partition evolution in spark